servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xPublishController.java

```diff
@@ -50,17 +57,28 @@ public IStatus canPublish() {
 		return Status.OK_STATUS;
 	}
 
+	protected IPublishBehaviour getJMXPublisher() {
+		return publisher2;
+	}
+	
+	protected IPublishBehaviour getPublisher(IModule[] module) {
+		return getJMXPublisher();
+	}
+	
 	/*
 	 * (non-Javadoc)
 	 * @see org.jboss.ide.eclipse.as.wtp.core.server.behavior.IPublishController#canPublishModule(org.eclipse.wst.server.core.IModule[])
 	 */
 	@Override
 	public boolean canPublishModule(IModule[] module) {
-		for (IModule m : module) {
-			if (!m.getModuleType().getId().equals("fuse.camel") && !m.getModuleType().getVersion().equals("1.0")) {
-				return false;
-			}
-		}
+		// This should be handled by plugin.xml really. 
+//		for (IModule m : module) {
+//			if (!m.getModuleType().getId().equals("fuse.camel") && !m.getModuleType().getVersion().equals("1.0")) {
+//				return false;
```

Comment:
probably doesn't need to be there... but doesn't hurt anything. As a rule,  "try it and see".  ie, try it, and see if any of the primary usecases break.